### ISSUE TITLE
IconMan background color not honored when not holding an icon

### ISSUE THREAD

**Issue by bender647** [fvwmorg/fvwm#17]
My desire is to have an FvwmIconMan instance on the root window, and blend into the root window by taking on it's color.  I'd prefer to use colorsets so the change is immediate when I switch themes.

My problem is, when there is no minimized window, the background of the icon area is gray, never a color I specify.

I trimmed down a minimal configuration file that demonstrates this. I'm using fvwm 2.6.6.

```
Exec xsetroot -solid "rgb:87/ce/fa"

DestroyModuleConfig FvwmIconMan

# Colorset config - this doesn't seem to work
CleanupColorsets
Colorset 0 Foreground rgb:87/ce/fa, Background rgb:87/ce/fa
*FvwmIconMan: PlainColorset 0
*FvwmIconMan: IconColorset 0
*FvwmIconMan: FocusColorset 0
*FvwmIconMan: IconAndSelectColorset 0
*FvwmIconMan: SelectColorset 0

## Non-colorset config - this seems to work
#*FvwmIconMan: Foreground rgb:87/ce/fa
#*FvwmIconMan: Background rgb:87/ce/fa

*FvwmIconMan: ButtonGeometry 25x25
*FvwmIconMan: DontShow Class=FvwmIconMan
*FvwmIconMan: DrawIcons Always
*FvwmIconMan: Format ""
*FvwmIconMan: ManagerGeometry 10x1
*FvwmIconMan: ReliefThickness 0
*FvwmIconMan: ShowOnlyIcons True
*FvwmIconMan: Title ""

Style FvwmIconMan BorderWidth 1, HandleWidth 0
Style FvwmIconMan StaysOnBottom, NoHandles, NoTitle, Sticky

Module FvwmIconMan

Style * NoIcon, MiniIcon mini-term.xpm

Mouse 1 I A Iconify Off
Mouse 0 4 A Iconify
```

**Comment by somiaj** [comment]
I have also ran into this issue and think I have tracked down the cause.

I had the issue that if using *FvwmIconMan: Colorset N, it was not being used to draw the background of the area outside of the buttons. After looking around I discovered that if you didn't set the default Colorset and just used Background and Foreground it honored that.

I was able to use this to track down the area that was drawing the empty part of the buttons and see that a condition to draw if the Colorset was missing. I copied the format of the function from draw_buttons_background to draw the Colorset and that worked when I tested it (patch attached).

For a fix in 2.6.7, just don't use a Default Colorset and use Background and Foreground instead. Then use colorsets for the other aspects. This at least works.


````
--- a/modules/FvwmIconMan/xmanager.c
+++ b/modules/FvwmIconMan/xmanager.c
@@ -1097,7 +1097,9 @@ static void clear_empty_region(WinManager *man)
                rects[0].x, rects[0].y, rects[0].width, rects[0].height,
                rects[1].x, rects[1].y, rects[1].width, rects[1].height);
 
-  if (CSET_IS_TRANSPARENT_PR_PURE(man->colorsets[DEFAULT]))
+  int cset = man->colorsets[DEFAULT];
+
+  if (CSET_IS_TRANSPARENT_PR_PURE(cset))
   {
     for(n=0; n < num_rects; n++)
     {
@@ -1109,6 +1111,12 @@ static void clear_empty_region(WinManager *man)
            }
     }
   }
+  else if (cset >= 0)
+  {
+    SetRectangleBackground(theDisplay, man->theWindow, rects[n].x, rects[n].y,
+                           rects[n].width, rects[n].height, &Colorset[cset],
+                           Pdepth, man->backContext[DEFAULT]);
+  }
   else
   {
     XFillRectangles(theDisplay, man->theWindow,
````

**Comment by ThomasAdam** [comment]
So this issue is just one of initialisation.  What should happen is the ordering; all colorsets should be initalised to the same value and then let them diverge when settings change them.

A fix has been pushed for this.
